### PR TITLE
Optionally use sudo when running docker build/push

### LINKED ISSUE #1548: Add optional (or not) sudo to docker commands
By default `kontena stack build` runs `docker` without sudo, I suggest to add sudo to every docker command, that way user running kontena doesn't needs to be in docker group (security risk)

### PR DESCRIPTION
**Pull request by kke**
Fixes #1548

### CODE REVIEW

**Comment by kke** [comment]
Is it actually needed for build? Or just push?

**Comment by nevalla** [comment]
I think both. Both commands are executing `docker` command which might require sudo

**Comment by jalberto** [comment]
@kke it's also needed for docker login adn pretty much any docker command (if some docker  subcommands use sudo and other doesn't it will create 2 different contexts)

In other hand, will be handy if --sudo is aglobal option, that way it can be used in an alias

    alias kontena="kontena --sudo"

**Comment by kke** [comment]
@jalberto The build and push here are the only docker commands that the cli is running.

**Comment by jakolehm** [comment]
Why we need an option for sudo.. why not just use `sudo kontena ...`?

**Comment by jalberto** [comment]
@jakolehm because run a ruby gem as sudo is a bad practice for security reasons, also can cause problems with RVM/rbenv and similar tools

File: cli/lib/kontena/cli/stacks/build_command.rb [hunk 1]
```diff
@@ -11,6 +12,8 @@ class BuildCommand < Kontena::Command
     option ['--no-cache'], :flag, 'Do not use cache when building the image', default: false
     option ['--no-push'], :flag, 'Do not push images to registry', default: false
     option ['--no-pull'], :flag, 'Do not attempt to pull a newer version of the image', default: false
+    option ['--[no-]sudo'], :flag, 'Run docker using sudo', hidden: Kontena.on_windows?, environment_variable: 'KONTENA_SUDO', default: false
```
Comment (jnummelin) [inline cli/lib/kontena/cli/stacks/build_command.rb]: Why do we need the `no-sudo` option when the default is false?

Comment (kke) [inline cli/lib/kontena/cli/stacks/build_command.rb]: I was hoping someone would ask :)

If you have `KONTENA_SUDO=true` but want to run the command without sudo, you would use `--no-sudo` which overrides the environment variable.

Comment (jnummelin) [inline cli/lib/kontena/cli/stacks/build_command.rb]: That's a bit weird combo to use I think. :) But it should not harm to have that option.

**Pull request merged**